### PR TITLE
Support non-JSON ServerRequestInterface

### PR DESCRIPTION
**Pull request by dsavina**
In #598, use of `ServerRequestInterface::getParsedBody()` has been replaced with inlined parsing of `RequestInterface::getBody()`.

Later, it has been reintroduced (cf. #715), but for JSON requests only.

Therefore, this Pull Request aims to restore use of `getParsedBody()` when available, in fallback case as well (commonly `application/x-www-form-urlencoded`).

### CODE REVIEW

**Comment by spawnia** [comment]
Please add a test to ensure this actually fixes a reproducible issue and we do not cause a regression later.

File: src/Server/Helper.php [hunk 1]
```diff
@@ -558,6 +556,22 @@ protected function decodeJson(string $rawBody)
         return $bodyParams;
     }
 
+    /**
+     * @return array
+     *
+     * @throws RequestError
+     */
+    protected function decodeContent(string $rawBody, string $contentType)
```
Comment (spawnia) [inline src/Server/Helper.php:564]:
```suggestion
    protected function decodeContent(string $rawBody, string $contentType): array
```

File: src/Server/Helper.php [hunk 2]
```diff
@@ -558,6 +556,22 @@ protected function decodeJson(string $rawBody)
         return $bodyParams;
     }
 
+    /**
+     * @return array
```
Comment (spawnia) [inline src/Server/Helper.php:560]:
```suggestion
     * @return array<string, mixed>
```

**Comment by dsavina** [comment]
Thanks for the feedback, I updated the code as requested.

I actually had an issue using Symfony's [`PsrHttpFactory`](https://github.com/symfony/psr-http-message-bridge/blob/main/Factory/PsrHttpFactory.php), which, for some reason, ends up providing a `ServerRequest` with an empty `body` but a valid `parsedBody`.

Although this could be perceived as an issue on Symfony's side, considering that, in the case of an URL-encoded body:
- Method `Helper::parseHttpRequest()` relies on global variable `$_POST` (i.e. already parsed)
- Test method `RequestParsingTest::parseRawFormUrlencodedRequest()` indicates that reading from `php://input` should be prohibited

, I guess this behavior is nevertheless considered pretty standard, isn't it?

Lastly, in order to support these inconsistencies between `body` and `parsedBody`, I could propose this instead:
```php
                if ($request instanceof ServerRequestInterface) {
                    $bodyParams = $request->getParsedBody();
                }

                $bodyParams ??= $this->decodeContent((string) $request->getBody(), $contentType[0]);
```
Would that be preferable?

**Comment by spawnia** [comment]
> Would that be preferable?

If you can create a test case that fails with the current implementation and passes with that one, sure.

**Comment by spawnia** [comment]
Thanks for the PR and the thorough tests

**Comment by dsavina** [comment]
Hello @spawnia, do you have some insight regarding when this could be part of a new release, by any chance?

We will need this fix for our package [thecodingmachine/graphqlite-bundle](https://github.com/thecodingmachine/graphqlite-bundle). If it's not something we can expect in the near future (which I would totally understand, I don't mean to push my personal agenda), I guess we could implement some temporary fix on our side. I'd still prefer not to if it's something we can avoid, though.

**Comment by spawnia** [comment]
Backported and releaed with https://github.com/webonyx/graphql-php/releases/tag/v14.11.2

**Comment by dsavina** [comment]
Nice, thank you very much!

**Pull request merged**